dev-cpp/sourcetrail/sourcetrail-2018.3.13.ebuild

```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+MULTILIB_COMPAT=( abi_x86_64 )
+
+inherit desktop gnome2-utils xdg-utils
+
+DESCRIPTION="A cross-platform source explorer for C/C++ and Java"
+HOMEPAGE="https://www.sourcetrail.com/"
+SRC_URI="https://www.sourcetrail.com/downloads/${PV}/linux/64bit -> ${P}.tar.gz"
+LICENSE="Sourcetrail"
+
+SLOT="0"
+KEYWORDS="~amd64"
+DEPEND="dev-util/patchelf"
+S="${WORKDIR}/Sourcetrail"
+QA_PREBUILT="*"
+
+src_install() {
+	for f in "${S}/Sourcetrail" "${S}/sourcetrail_indexer"; do
+		rpath=$(patchelf --print-rpath "${f}" 2>/dev/null)
+		[[ ${rpath} =~ (.*:)?\$\$ORIGIN[^:]*(:.*)? ]] || continue
+		patchelf --set-rpath "${BASH_REMATCH[1]}${BASH_REMATCH[2]#:}" "${f}" || die
+	done
+
+	insinto /opt/sourcetrail
+	doins -r *
+	exeinto /opt/sourcetrail
+	doexe Sourcetrail
+	doexe sourcetrail_indexer
+	doexe Sourcetrail.sh
+	dodir /usr/bin
+	cat <<-EOF >"${D}"/usr/bin/sourcetrail || die
+		#! /bin/sh
+		exec /opt/sourcetrail/Sourcetrail.sh "\$@"
+	EOF
+	fperms +x /usr/bin/sourcetrail
+	local size
+	for size in 48 64 128 256 512; do
+		newicon -s ${size} "${S}/setup/share/icons/hicolor/${size}x${size}/apps/sourcetrail.png" \
```

Comment:
Actually the icons and desktop file is installed in /opt/sourcetrail/... etc, but I think for desktop systems to pick them up, they need to be installed in /usr/share/icons and /usr/share/applications accordingly, that is when the `newicon` and `newmenu` functions do. But I have removed the ${S} from both, since that is not necessary.

Reply:
I was rather asking if there's a point in installing them to /opt.

Reply:
I don't know if the application itself uses these icons, should I try removing them from /opt and then only installing them is `newicon`? Or isn't it worth the effort?

Reply:
Please do check.